applications/PfemFluidDynamicsApplication/custom_processes/remove_mesh_nodes_for_fluids_cut_pfem_process.hpp

```diff
@@ -0,0 +1,1369 @@
+			Edges[0] = std::sqrt(SquaredLength);
+			FirstEdgeNode[0] = 0;
+			SecondEdgeNode[0] = 1;
+			if (eElement[0].Is(RIGID) && eElement[1].Is(RIGID))
+			{
+				wallLength = Edges[0];
+			}
+			bool inletElement = false;
+			if (eElement[0].Is(PFEMFlags::EULERIAN_INLET) || eElement[1].Is(PFEMFlags::EULERIAN_INLET) || eElement[2].Is(PFEMFlags::EULERIAN_INLET))
+			{
+				inletElement = true;
+			}
+			SizeType counter = 0;
+			for (SizeType i = 2; i < eElement.size(); i++)
+			{
+				for (SizeType j = 0; j < i; j++)
+				{
+					noalias(CoorDifference) = eElement[i].Coordinates() - eElement[j].Coordinates();
+					SquaredLength = CoorDifference[0] * CoorDifference[0] + CoorDifference[1] * CoorDifference[1];
+					counter += 1;
+					Edges[counter] = std::sqrt(SquaredLength);
+					FirstEdgeNode[counter] = j;
+					SecondEdgeNode[counter] = i;
+					if (eElement[i].Is(RIGID) && eElement[j].Is(RIGID) && Edges[counter] > wallLength)
+					{
+						wallLength = Edges[counter];
+					}
+				}
+			}
+
+			////////  to compare the triangle height to wall edge length /////////
+			for (SizeType i = 0; i < eElement.size(); i++)
+			{
+				if (eElement[i].IsNot(RIGID) && eElement[i].IsNot(TO_ERASE) && eElement[i].IsNot(SOLID) && eElement[i].IsNot(ISOLATED))
+				{
+					double height = elementVolume * 2.0 / wallLength;
+
+					//////it is evident when a freesurface particle in touch with wall is erased --> reduce the safety coeff
+					if (eElement[i].Is(FREE_SURFACE))
+					{
+						NodeWeakPtrVectorType &neighb_nodes = eElement[i].GetValue(NEIGHBOUR_NODES);
+						SizeType countRigid = 0;
+						SizeType countFreeSurface = 0;
+						for (NodeWeakPtrVectorType::iterator nn = neighb_nodes.begin(); nn != neighb_nodes.end(); nn++)
+						{
+							if ((nn)->Is(RIGID) || (nn)->Is(SOLID))
+							{
+								countRigid++;
+							}
+							if ((nn)->Is(FREE_SURFACE) && (nn)->IsNot(RIGID) && (nn)->IsNot(SOLID))
+							{
+								countFreeSurface++;
+							}
+						}
+						if ((countRigid + countFreeSurface) == neighb_nodes.size() && countRigid > 0)
+						{
+							safetyCoefficient2D = 0.25;
+						}
+					}
+
+					////// if the node is very close to the wall is erased in any case
+					if (height < (0.5 * safetyCoefficient2D * wallLength))
+					{
+						eElement[i].Set(TO_ERASE);
+						inside_nodes_removed++;
+
+						if (inletElement)
+						{
+							nodes_removed_inlet_zone++;
+						}
+
+						const SizeType propertyIdNode = eElement[i].FastGetSolutionStepValue(PROPERTY_ID);
+						if (propertyIdNode != principalModelPartId) // this is to conserve the number of nodes of the smaller domain in case of a two-fluid analysis
+						{
+							mrRemesh.Info->BalancePrincipalSecondaryPartsNodes += -1;
+						}
+					}
+				}
+			}
+
+			bool longDamBreak = false; // to attivate in case of long dam breaks to avoid separeted elements in the water front
+			if (longDamBreak == true)
```

Comment:
```suggestion
			if (longDamBreak)
```

Reply:
besides, the `If` will never be activated since we just defined this variable to false right?